core/modules/wpublish/QueriesAndChunks.cc

```diff
@@ -462,17 +463,32 @@ QueriesAndChunks::removeQueryFrom(QueryId const& qId,
         }
     }
 
-    for(auto const& task : taskList) {
-        auto taskSched = task->getTaskScheduler();
-        if (taskSched != nullptr && (taskSched == sched || sched == nullptr)) {
-            // removeTask will only return the task if it was found on the
-            // queue for its scheduler and removed.
-            auto removedTask = taskSched->removeTask(task);
-            if (removedTask != nullptr) {
-                removedList.push_back(removedTask);
+    auto moveTasks = [this, &sched, &taskList, &removedList](bool moveRunning) {
+        std::vector<wbase::Task::Ptr> taskListNotRemoved;
+        for(auto const& task : taskList) {
+            auto taskSched = task->getTaskScheduler();
+            if (taskSched != nullptr && (taskSched == sched || sched == nullptr)) {
+                // removeTask will only return the task if it was removed from the
+                // queue for its scheduler.
+                auto removedTask = taskSched->removeTask(task, moveRunning);
+                if (removedTask != nullptr) {
+                    removedList.push_back(removedTask);
+                } else {
+                    taskListNotRemoved.push_back(task);
+                }
```

Comment:
Does `removeTaks(task)` always return `task` if not `nullptr`? The code in the `if` sort of implies that but it's somewhat confusing me to think about what happens if `task` and `removedTask` are not the same. Would it be simpler if `removeTaks(task)` just returned bool?